### PR TITLE
Remove references to FUTURE SEQUANA target

### PR DESCRIPTION
**Pull request by MarceloSalazar**
5.11 introduces a non-standard mechanism to add a feature, by creating a new target
This is not something we should be doing, however, it was agreed with the SPM team this is a temporary solution and will be fixed in a future release.

Also added example on how to mock mbedls to detect the PSA target and how to run tests

** READY TO MERGE **

### CODE REVIEW

File: docs/porting/psa/spm.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 Secure Partition Manager (SPM) is a part of the PSA Firmware Framework that is responsible for isolating software in partitions, managing the execution of software within partitions and providing interprocessor communication (IPC) between partitions.
 
-For more information about SPM, please refer to [the SPM overview page](../docs/development/apis/mbed-psa.html).
+For more information about SPM, please refer to [the SPM overview page](../apis/psa-api.html).
 
 <span class="notes">This page gives guidelines for silicon partners wishing to have Secure Partition Manager capabilities.</span>
```
Comment (iriark01) [inline docs/porting/psa/spm.md]: Why "wishing to have" rather than "adding"?

File: docs/porting/psa/spm.md [hunk 2]
```diff
@@ -20,10 +20,10 @@ When adding a new target, a new root target node should be added to the `mbed-os
 These is demonstrated in the example below:
```
Comment (iriark01) [inline docs/porting/psa/spm.md]: "These is"; please fix

File: docs/porting/psa/spm.md [hunk 3]
```diff
@@ -258,3 +258,13 @@ After finalizing the porting, execute the following tests:
 - **tests-mbed_hal-spm:** This test will make sure the porting of the memory protection (*spm_hal_memory_protection_init()* implementation) makes the correct partitioning between secure RAM/Flash and nonsecure RAM/Flash.
 
 We recommended you leave the memory protection part (*spm_hal_memory_protection_init()* implementation) to the end of the porting. First, implement and test other HAL functions. After these tests pass, implement *spm_hal_memory_protection_init()*, and run the entire test suite again, including the memory protection related tests.
```
Comment (iriark01) [inline docs/porting/psa/spm.md]: Please use '' instead of * * to show function names.

File: docs/porting/psa/spm.md [hunk 4]
```diff
@@ -258,3 +258,13 @@ After finalizing the porting, execute the following tests:
 - **tests-mbed_hal-spm:** This test will make sure the porting of the memory protection (*spm_hal_memory_protection_init()* implementation) makes the correct partitioning between secure RAM/Flash and nonsecure RAM/Flash.
 
 We recommended you leave the memory protection part (*spm_hal_memory_protection_init()* implementation) to the end of the porting. First, implement and test other HAL functions. After these tests pass, implement *spm_hal_memory_protection_init()*, and run the entire test suite again, including the memory protection related tests.
+
+This is an example on how to run SPM tests for an Mbed OS platform (tipicaly a nonsecure core of PSA target):
```
Comment (iriark01) [inline docs/porting/psa/spm.md]: Please fix this sentence

File: docs/porting/psa/spm.md (hunk 3 above)
Comment (iriark01) [reply docs/porting/psa/spm.md]: Also, tests don't "make sure" - that implies fixing. They check.

File: docs/porting/psa/spm.md [hunk 5]
```diff
@@ -250,11 +250,21 @@ Processor access    |Secure RAM        |Secure FLASH|Nonsecure RAM      |Nonsecu
...
 
-We recommended you leave the memory protection part (*spm_hal_memory_protection_init()* implementation) to the end of the porting. First, implement and test other HAL functions. After these tests pass, implement *spm_hal_memory_protection_init()*, and run the entire test suite again, including the memory protection related tests.
+We recommended you leave the memory protection part (`spm_hal_memory_protection_init()` implementation) to the end of the porting. First, implement and test other HAL functions. After these tests pass, implement `spm_hal_memory_protection_init()`, and run the entire test suite again, including the memory protection related tests.
+
+This example shows how to run SPM tests for a PSA target with a nonsecure core:
+
+```
+mbed test -t GCC_ARM -m SPM_NONSECURE_CORE_PSA -n mbed-os-tests-psa-spm_smoke -v
+```
+
```
Comment (iriark01) [inline docs/porting/psa/spm.md]: There are two lines of code. What's the difference between them? Needs to be called out.

Comment (MarceloSalazar) [reply docs/porting/psa/spm.md]: These are two separate test commands. It's ok to combine them within the same code context if you prefer.

**Pull request merged**